src/expr/src/vector_op/agg/filter.rs

```diff
@@ -0,0 +1,213 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use risingwave_common::array::{ArrayBuilderImpl, DataChunk};
+use risingwave_common::buffer::Bitmap;
+use risingwave_common::types::DataType;
+
+use super::aggregator::Aggregator;
+use super::BoxedAggState;
+use crate::expr::ExpressionRef;
+use crate::Result;
+
+#[derive(Clone)]
+pub struct Filter {
+    filter: ExpressionRef,
+    child: BoxedAggState,
+}
+
+impl Filter {
+    pub fn new(filter: ExpressionRef, child: BoxedAggState) -> Self {
```

Comment:
Call it `child` is slightly misleading, which looks like the filter is applied to the output of the agg 😄. 

I'm thinking about something like `agg.with_filter(filter)`, but maybe simply adding some docs to `Filter` is enough.